lib/recog/db.rb

```diff
@@ -13,13 +13,34 @@ class DB
   #   against strings that make sense for the {#match_key}
   attr_reader :fingerprints
 
-  # @return [String] Taken from the `fingerprints/matches` element, or
+  # @return [String] Taken from the `fingerprints/matches` attribute, or
   #   defaults to the basename of {#path} without the `.xml` extension.
   attr_reader :match_key
 
+  # @return [String] Taken from the `fingerprints/protocol` attribute, or
+  #   defaults to an empty string
+  attr_reader :protocol
+
+  # @return [String] Taken from the `fingerprints/database_type` attribute
```

Comment:
The only place I spotted `type` is in the cucumber testing XML which isn't covered by spec. I've fixed it there now.